Fix trace headers for existing cloud events

yaron2 opened the pull request:
Closes https://github.com/dapr/dapr/issues/6800.

daixiang0 commented:
Codes looks good, could we add tests to cover it?

ItalyPaleAle reviewed (dismissed):
LGTM but It would be nice if this code (that adds trace headers) could be moved to a function, for example in the universal package, so it’s not repeated twice (and we don’t risk it drifts in the future)

yaron2 commented:
> LGTM but It would be nice if this code (that adds trace headers) could be moved to a function, for example in the universal package, so it’s not repeated twice (and we don’t risk it drifts in the future)

Changed, addressed two feedbacks with one commit.
